firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/Containers/EnumMap.txt

```diff
@@ -1,303 +1,355 @@
-      THEN('The input method returns ok status for second key') 
-      THEN('The operator method returns expected value for third key') 
-      THEN('The input method returns ok status for third key') 
-      THEN('The max_size method reports, EnumMap has capacity 4')
       THEN('After the input, the size method reports size as 3 ') 
-      THEN('The avaliable method reports that 1 key-value pair are avaliable') 
-      THEN('The full method reports that the EnumMap is not completely filled') 
+      THEN('The avaliable method reports that 1 item is avaliable') 
+      THEN('The full method reports that the EnumMap is not full') 
       THEN('The empty method reports that the EnumMap is non-empty') 
-      THEN('The has method returns true for 3 keys') 
+      THEN('The erase method reports ok status for 3rd key ') 
+      THEN('The has method returns true for 1,2 and 3 keys') 
+      THEN('The has method returns false for [-1000]to[1000] keys')
 
+SCENARIO('The input method in EnumMap works correctly for struct and pointers') 
   GIVEN('An empty EnumMap with capacity 4') 
-    WHEN('The input method is called, where values are elemets of struct') 
-      THEN('The operator method returns expected value for first key') 
-      THEN('The input method returns ok status for first key') 
-      THEN('The operator method returns expected value for second key') 
-      THEN('The input method returns ok status for third key') 
-      THEN('The operator method returns expected value for third key') 
-      THEN('The input method returns ok status for third key') 
+    WHEN('The input method is called thrice for 1,2 and 3 key, where values are elemets of struct 121, 2 and 898 respectively') 
+      THEN('The operator method returns expected value for 1st key') 
+      THEN('The input method returns ok status for 1st item') 
+      THEN('The operator method returns expected value for 2nd key') 
+      THEN('The input method returns ok status for 2nd item') 
+      THEN('The operator method returns expected value for 3rd key') 
+      THEN('The input method returns ok status for 3rd item') 
       THEN('The max_size method reports, EnumMap has capacity 4') 
       THEN('After the input, the size method reports size as 3') 
-      THEN('The avaliable method reports that 1 key-value pair are avaliable') 
-      THEN('The full method reports that the EnumMap is not completely filled') 
+      THEN('The avaliable method reports that 1 item is avaliable') 
+      THEN('The full method reports that the EnumMap is not full') 
       THEN('The empty method reports that the EnumMap is non-empty') 
-      THEN('The has method returns true for 3 keys') 
+      THEN('The erase method reports ok status for 3rd key ') 
+      THEN('The has method returns true for 1,2 and 3 keys') 
+      THEN('The has method returns false for [-1000]to[1000] keys')
+      THEN('Changing the value of struct elements does not change values of key 1,2 and 3')
```

Comment:
The struct fields are correctly set by direct access using the [] operator:
```
map[1].value = 122
map[2].value = 3
map[3].value = 899
assert(map[1].value == 122)
```

The map values are correctly set by assignment on the [] operator:
```
TestStruct val4{122}
TestStruct val5{3}
TestStruct val6{899}
map[1] = val4
map[2] = val5
map[3] = val6
assert(map[1].value == 122)
```